lib/fizzy/execute.cpp

```diff
@@ -1034,39 +1034,36 @@ ExecutionResult execute(Instance& instance, FuncIdx func_idx, const Value* args,
         }
         case Instr::i32_div_s:
         {
-            auto const rhs = stack[0].as<int32_t>();
-            auto const lhs = stack[1].as<int32_t>();
+            const auto rhs = stack.pop().as<int32_t>();
+            const auto lhs = stack.top().as<int32_t>();
             if (rhs == 0 || (lhs == std::numeric_limits<int32_t>::min() && rhs == -1))
                 goto trap;
-            binary_op(stack, div<int32_t>);
+            stack.top() = div(lhs, rhs);
```

Comment:
Is this much better?

`binary_op` is doing the same, and also with this change (and similar changes below) we're somewhat incosistent in handling traps: `call` that resulted in a trap leaves arguments on stack, other instructions that trap consume (some of) their arguments.

Reply:
This is simpler in a sense the stack items are accessed only once. Compiler can generate better code and the size of `execute()` is smaller.
Secondly, this removes the usage of `stack[]` syntax so `operator[]` overloading can be dropped.

The rest I don't really care about. If this can be implemented in more readable form with the above restrictions I'm open for suggestions.

Reply:
BTW, there are some other `div`/`rem` instructions remaining using `binary_op()`. I wasn't pushing for consistency here, but they can be revisit.

Other approach is to wrap division instruction with a bigger helpers returning `bool` for trap (as we have in other cases).

Reply:
I'm fine with leaving it like this.